Async client

leits opened the pull request.

leits committed:
Fix contributing doc

leits committed:
Add _async client

leits committed:
Create utils.py

leits committed:
Import async client to deta init

leits committed:
Add async extra

leits committed:
Add async tests

leits commented:
@aavshr, +1 point to review this PR:
https://github.com/rickh94/ODetaM/issues/3

aavshr commented on Makefile:
```diff
@@ -2,10 +2,10 @@
 
 test: # Run Unit Test
-	python -m unittest tests
+	pytest tests
```
we do not use any other dependencies for tests currently and the CI workflow also does not install any dependencies and just runs a single file: https://github.com/deta/deta-python/blob/master/.github/workflows/pull_request.yml#L24-L31

we need to decide internally if we should use `pytest`, if yes, we also need to update the other tests to use `pytest` so that it's consistent and update the PR workflow.

leits replied:
ok

pytests is recommended way to test aiohttp. And it makes the developer experience better.

But if you decided not to use it, there is a way for testing aiohttp with unittest and I can rewrite tests:
https://docs.aiohttp.org/en/stable/testing.html#unittest

aavshr replied:
`pytests` should be fine, we'll use it from now on for tests. You don't have to update the github workflow or the other tests. I will do that in a different PR.

aavshr commented on deta/_async/client.py:
```diff
@@ -0,0 +1,167 @@
+from deta.base import FetchResponse, Util
+
+
+class AsyncDeta:
```
I don't think we should have an `AsyncDeta` class as it is redundant and does not add anything new/different to the `Deta` class. We can just do: 
```python
from deta import Deta

async_base = Deta("project_key").AsyncBase("base_name")
```

leits replied:
Make sense. I'll change it.

leits replied:
Removed.

leits replied:
Updated. Check it, please.

aavshr reviewed (commented):
thank you for the PR @leits  we appreciate it very much. 

Sorry for taking so much time to review this. Commented a couple of things.

leits commented:
> thank you for the PR @leits we appreciate it very much.
> 
> Sorry for taking so much time to review this. Commented a couple of things.

Thanks for the review, @aavshr.
No problem. I'll remove redundant AsyncDeta class and will wait to your decision about pytests 👌

leits committed:
Remove redundant AsyncDeta class

leits committed:
Return accidentally removed os import

aavshr commented on tests/test_async.py:
```diff
@@ -0,0 +1,241 @@
+from deta.base import FetchResponse
+import os
+import pytest
+from deta import AsyncDeta
```
`from deta import Deta`

aavshr commented on tests/test_async.py:
```diff
@@ -0,0 +1,241 @@
+    assert BASE_NAME
+
+    deta = AsyncDeta(PROJECT_KEY)
+    db = deta.Base(BASE_NAME)
```
```py
deta = Deta(PROJECT_KEY)
db = deta.AsyncBase(BASE_NAME)
```

aavshr reviewed (commented):
thanks @leits , just two small changes needed for the tests and then we can merge. :tada:

leits committed:
Remove using AsyncDeta in tests

leits commented:
> thanks @leits , just two small changes needed for the tests and then we can merge. 🎉

Damn, I changed that but forgot to commit 😅
Goot catch! Fixed!

aavshr reviewed (approved):
thanks @leits .
